### ISSUE TITLE
Mythos end-of-turn damage even when you play his cards

### ISSUE THREAD

**Issue by MichaelLorenzMitchell** [SotMSteamMods/CauldronMods#1264]
**Describe the bug**
Playing optional cards via Dangerous Investigation did not prevent the end-of-turn damage from Mythos.

**What does it affect**
Card(s): Dangerous Investigation
Deck(s): Mythos

**To Reproduce**
1. Pyre, Gyrosaur, Magnificent Mara, Drift
2. In play: Mythos - Dangerous Investigation (no tokens). Pyre - Containment Breach, Thermonuclear Core. Gyro - none. Mara - Dowsing Crystal, Post Hypnotic Cue (on self). Drift - (Drift Track=R), Throwing Shard. Catchwater Harbor - Cervantes Club, Frenzied Onlookers
3. Mythos plays Faithful Proselyte, Mara uses dowsing crystal (non-breaking form) to allow Pyre to do damage to Faithful Proselyte. Using Mythos's card, players Faithful Proselyte, Otherworldly Alignment. Clockwork Revenant, and Hallucinated Horror. Using Dangerous Investigation, players play Ritual Site. A token is added to Dangerous Investigation.
4. Mythos does 3 infernal damage to all heroes using Dangerous Investigation.
 
**Expected behavior**
Expected Mythos to deal damage to (H)-"Number of optionally played cards" = 4-4 = 0 heroes, instead he did damage to all heroes. Rewinding and playing fewer of his cards did not change the behavior

**Screenshots**
![DangerousInvestigationDamage](https://user-images.githubusercontent.com/83606037/116914661-01d79e80-ac19-11eb-9639-fa7daa893e15.png)

**Logs**
[log.txt](https://github.com/SotMSteamMods/CauldronMods/files/6416774/log.txt)
[output_log.txt](https://github.com/SotMSteamMods/CauldronMods/files/6416775/output_log.txt)

**Comment by wrhyme29** [comment]
I can dig into it but its worth noting that you were playing on v0.4.2 which is 2 releases old, which might have caused the issue. Can you try to replicate on the latest version?

**Comment by MichaelLorenzMitchell** [comment]
To my knowledge, I'm on the workshop version I thought that updated automatically. Before posting, I did a search for this issue in this tracker to see if it had already been seen/addressed in a dev build, but didn't find any mention of it.

**Comment by wrhyme29** [comment]
It should be updating automatically now, but it didn't used to be. Unsure why yours would still show as an older version though (the output log is what tells me that). I would recommend unsubscribing a re-subscribing to the Workshop should force it to update to the latest version

**Comment by MichaelLorenzMitchell** [comment]
Looks like v0.4.2 is the version I had of the pre-workshop mod. I had thought by renaming the folder with the mod in it that it would break the connection and I would get the workshop version. My betas setting was "None" (I changed it away from Alpha Mod Support when I was trying to get the workshop version).

I've now moved CauldronMods out of the Mods folder. I changed the beta setting to "mods - Alpha mod support" and un/resubscribed.

I'm not sure what version I'm on now (output.log no longer includes it) but Dangerous Investigation now works correctly.
